cell_classification/segmentation_data_prep_test.py

```diff
@@ -33,9 +49,165 @@ def test_get_image():
         assert not np.array_equal(CD8_img, CD4_img)
 
 
+def prepare_test_data_folders(
+    num_folders, temp_dir, selected_markers, random=False, scale=1.0
+):
+    data_folders = []
+    for i in range(num_folders):
+        folder = os.path.join(temp_dir, "fov_1" + str(i))
+        os.mkdir(folder)
+        data_folders.append(folder)
+        for marker in selected_markers:
+            if random:
+                img = np.random.rand(256, 256) * scale
+            else:
+                img = np.ones([256, 256])
+            imwrite(
+                os.path.join(temp_dir, "fov_1" + str(i), marker + ".tiff"),
+                img,
+            )
+    return data_folders
+
+
+def test_calculate_normalization_matrix():
+
+    # instantiate data_prep, conversion_matrix and markers
+    data_prep = prep_object()
+    conversion_matrix = prepare_conversion_matrix()
+    selected_markers = list(conversion_matrix.columns)
+    selected_markers.remove("cluster_labels")
+
+    # check normalization_dict for different stochastic images
+    for scale in [0.5, 9.132]:
+
+        # create temporary folders with data and do tests
+        with tempfile.TemporaryDirectory() as temp_dir:
+            data_folders = prepare_test_data_folders(
+                5, temp_dir, selected_markers, random=True, scale=scale
+            )
+            norm_dict = data_prep.calculate_normalization_matrix(
+                data_folders=data_folders,
+                normalization_dict_path=os.path.join(temp_dir, "norm_dict_test.json"),
+                normalization_quantile=0.99,
+                selected_markers=selected_markers,
+            )
+
+            # check if the normalization_dict has the correct values for stochastic images
+            for marker in norm_dict.keys():
+                assert norm_dict[marker] - 1 / (0.5 * scale) < 0.001
+
+            # check if the normalization_dict is correctly written to the json file
+            norm_dict_loaded = json.load(
+                open(os.path.join(temp_dir, "norm_dict_test.json"))
+            )
+            assert norm_dict_loaded == norm_dict
+
+            # check if the normalization_dict has the correct keys
+            for marker in selected_markers:
+                assert marker in norm_dict.keys()
+
+
+def test_make_tf_record():
+    data_prep = prep_object()
+    conversion_matrix = prepare_conversion_matrix()
+    selected_markers = list(conversion_matrix.columns)
+    selected_markers.remove("cluster_labels")
+    with tempfile.TemporaryDirectory() as temp_dir:
+
+        # create temporary folders with data and prepare normalization_dict
+        data_folders = prepare_test_data_folders(5, temp_dir, selected_markers)
+        norm_dict = data_prep.calculate_normalization_matrix(
```

Comment:
This is what's called an integration test. We already know that `calculate_normalization_matrix` works, because it has its own test function . So we don't need to recheck that the output is correct for every function that calls `calculate_normalization_matrix`. Instead, what we need to do is make sure the functions work together as expected. 

Therefore, calling `calculate_normalization_matrix` and checking the output against `data_prep.normalization_dict` is overkill. Instead, all we need to do is make sure that *whatever* normalization dict is loaded, that *same* normalization dict can be found added as an attribute. This allow us to further separate the tests from one another, and makes it shorter. 

I would make a simple normalization dict with two entries, save it, and make sure that same one is loaded.